context/app/static/js/components/entity-search/MetadataMenu/MetadataMenu.jsx

```diff
@@ -38,16 +39,34 @@ function NotebookMenuItem(props) {
   );
 }
 
-function MetadataMenu({ entityType }) {
+function MetadataMenu({ entityType, results }) {
   const lcPluralType = `${entityType.toLowerCase()}s`;
   const { selectedHits, createNotebook, closeMenu } = useMetadataMenu(lcPluralType);
   const menuID = 'metadata-menu';
 
   const { selectedRows } = useStore();
-  const errorMessage =
-    selectedRows.size > 10
-      ? `You have selected ${selectedRows.size} datasets. Workspaces currently only supports up to 10 datasets. Please unselect datasets.`
-      : null;
+
+  // array that contains all the datasets from current pagination where mapped_data_access_level === 'Protected'.
+  const filteredDataset = results?.hits?.items?.filter(
+    (item) => item?.fields?.mapped_data_access_level === 'Protected',
+  );
+
+  const containsProtectedDataset = filteredDataset?.some((item) => {
+    const filteredDatasetIds = item?.id;
+    // spreading out the selectedRows Set to convert to an array then checking to see if any contains mapped_data_access_level === 'Protected'.
+    return [...selectedRows].some((selectedRowId) => filteredDatasetIds.includes(selectedRowId));
+  });
+
+  let errorMessage;
```

Comment:
Do we want to show both messages if both cases were true?

Reply:
@tsliaw

Reply:
Both messages will show now.